ludwig/backend/ray.py

```diff
@@ -259,8 +267,11 @@ def tune_learning_rate_fn(
         initialize_pytorch(horovod=hvd)
 
         pipe = dataset.pipeline(shuffle=False, **data_loader_kwargs)
+
+        # Expensive blocking call
```

Comment:
Is this because of the new `len(dataset)`, or was this always an expensive blocking call?

Reply:
@justinxzhao It was always an expensive blocking call. I think calling the `count()` method on a Ray dataset or dataset pipeline is blocking and forces a full iteration over the dataset like Travis mentioned